### PR TITLE
Implement the new GraphQL elevationProfile type 

### PR DESCRIPTION
**Pull request by daniel-heppner-ibigroup**
This implements the type from #729. It also adds a test to ensure backwards compatibility is working properly.

### CODE REVIEW

**Comment by daniel-heppner-ibigroup** [comment]
@binh-dam-ibigroup this is ready for a review!

File: packages/core-utils/src/itinerary.ts [hunk 1]
```diff
@@ -329,30 +330,42 @@ export function getElevationProfile(
...
   steps.forEach(step => {
-    if (!step.elevation || step.elevation.length === 0) {
+    // Support for old REST response data (in step.elevation)
+    const stepElevationProfile =
+      step.elevationProfile ||
+      (Array.isArray(step.elevation) &&
+        step.elevation?.map<ElevationProfileComponent>(elev => ({
+          distance: elev.first,
+          elevation: elev.second
+        })));
```
Comment (binh-dam-ibigroup) [inline packages/core-utils/src/itinerary.ts:343]: Extract a method for this conversion, and reuse in the test.

**Review by binh-dam-ibigroup — COMMENTED** [review]
Elevation code changes look good, but did we need to touch all the package.json files? The reason I ask is there is only one `feat` commit for core-utils, and as a result, no new binaries will be generated for the other modified packages.

**Review by binh-dam-ibigroup — APPROVED** [review]
It is actually fine if no binaries are released for the other packages.

**Comment by daniel-heppner-ibigroup** [comment]
That's intentional! Just ran `update-internal-dependencies`

File: packages/core-utils/src/itinerary.ts [hunk 2]
```diff
@@ -329,30 +340,41 @@ export function getElevationProfile(
...
+      const elev = stepElevationProfile[i];
       if (previous) {
-        const diff = (elev.second - previous.second) * unitConversion;
+        const diff = (elev.elevation - previous.elevation) * unitConversion;
         if (diff > 0) gain += diff;
         else loss += diff;
       }
-      if (i === 0 && elev.first !== 0) {
+      if (i === 0 && elev.distance !== 0) {
         // console.warn(`No elevation data available for step ${stepIndex}-${i} at beginning of segment`, elev)
```
Comment (amy-corson-ibigroup) [inline packages/core-utils/src/itinerary.ts:366]: Is there a point to keeping this commented out `console.warn`?

**Review by amy-corson-ibigroup — APPROVED** [review]
Looks good! Even just changing `first` and `second` to `distance` and `elevation` is so much more clear

File: packages/core-utils/src/itinerary.ts (hunk 2 above)
Comment (daniel-heppner-ibigroup) [reply packages/core-utils/src/itinerary.ts:366]: IDK but it's not really relevant to this PR so I don't wanna touch it

**Pull request merged**